js/app.js

```diff
@@ -6,6 +6,182 @@ const navSlide = () => {
+  document.querySelectorAll('[data-see-live]').forEach((alink) => alink.setAttribute('href', projects.live_link));
+  document.querySelectorAll('[data-see-source]').forEach((alink) => alink.setAttribute('href', projects.repo));
+};
+
+function createCard(cardObject) {
+  const div = document.createElement('div');
+  div.className = 'card';
+  div.innerHTML = `<div class="container">
+  <div class="${cardObject.imageClass}"></div>
+  <div class="${cardObject.project}"><h2>${cardObject.name}</h2></div>
+  <div class="${cardObject.grid}">
+  <div class="html-css-rectangle"><p>HTML/CSS</p></div>
+  <div class="ruby-rails-rectangle"><p>Ruby on Rails</p></div>
+  <div class="javascript-rectangle"><p>Javascript</p></div>
+  <button data-modal-target="#modal" type="button" class="see-project">See this project</button>
+  <div class="arrow"></div>
+  </div>`;
+  return div;
+}
+
+function openModal(modal) {
+  if (modal == null) return;
+  modal.classList.add('active');
+  overlay.classList.add('active');
+  popup();
+}
+
+function closeModal(modal) {
+  if (modal == null) return;
+  modal.classList.remove('active');
+  overlay.classList.remove('active');
+}
+
+mobileProjects.forEach((project) => {
+  const card = createCard(project);
+  projectCards.appendChild(card);
+});
+
+openModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    scrollTop();
+    const modal = document.querySelector(button.dataset.modalTarget);
+    openModal(modal);
+  });
+});
+
+closeModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    const modal = button.closest('.modal');
+    closeModal(modal);
+  });
+});
+
+openModalButtons.forEach((button) => {
+  button.addEventListener('click', () => {
+    scrollTop();
+    const modal = document.querySelector(button.dataset.modalTarget);
+    openModal(modal);
+  });
+});
+
+const form = document.querySelector('[data-form]');
+const email = document.getElementById('list-email');
+const emailError = document.querySelector('#list-email + span.error');
+
+function showError() {
+  if (email.validity.valueMissing) {
+    emailError.textContent = 'Please enter an email address!';
+  } else if (email.validity.typeMismatch) {
+    emailError.textContent = 'Entered value needs to be an email address.';
+  } else if (email.validity.tooShort) {
+    emailError.textContent = `Email should be at least ${email.minlength} characters.`;
+  } if (email.value !== email.value.toLowerCase()) {
+    emailError.textContent = 'Please enter your email in lowercase';
+  }
+  emailError.className = 'error active';
+}
+
+email.addEventListener('input', () => {
+  if (email.validity.valid) {
+    emailError.textContent = '';
+    emailError.className = 'error';
+  } else {
+    showError();
+  }
+});
+
+form.addEventListener('submit', (event) => {
+  if (!email.validity.valid) {
+    showError();
+    event.preventDefault();
+  }
+});
```

Comment:
- [ ] Good job working on the validation. I can see you implement it here. However, when I continue typing the email, the error no longer shows, and the form submits anyway even when the email contains uppercase letters. Check the video I recorded. Kindly fix this.
https://www.loom.com/share/0aa539a20a004515b18aafa0f22a475f

- [ ] Also the error message should be placed close to the submit button as requested by the previous reviewer. Kindly fix it.

Reply:
Hey @phelian23 fixed both issues. Thanks for the quick review.

<img width="580" alt="Screenshot 2022-04-15 at 18 30 23" src="https://user-images.githubusercontent.com/40231363/163589957-18a51714-284f-49c2-907e-08758dcce231.png">